packages/@tailwindcss-vite/src/index.ts

```diff
@@ -239,106 +266,57 @@ export default function tailwindcss(): Plugin[] {
-            } as any
-          } else {
-            postcssConfig = {}
-          }
-          config.css = { postcss: postcssConfig }
-        }
-
-        // postcssConfig is no longer a string after the above. This test is to
-        // avoid TypeScript errors below.
-        if (typeof postcssConfig === 'string') {
-          return
-        }
-
-        if (!postcssConfig || !postcssConfig?.plugins) {
-          config.css = config.css || {}
-          config.css.postcss = postcssConfig || {}
-          config.css.postcss.plugins = [fixRelativePathsPlugin() as any]
-        } else {
-          postcssConfig.plugins.push(fixRelativePathsPlugin() as any)
-        }
-      },
-
       // Scan index.html for candidates
-      transformIndexHtml(html) {
-        let updated = scan(html, 'html')
+      transformIndexHtml(html, context) {
+        onFileChange(context.filename)
 
-        // In serve mode, if the generated CSS contains a URL that causes the
-        // browser to load a page (e.g. an URL to a missing image), triggering a
-        // CSS update will cause an infinite loop. We only trigger if the
-        // candidates have been updated.
+        let updated = scan(html, 'html')
         if (updated) {
-          updateCssModules(isSSR)
+          queueRebuild(isSSR)
         }
       },
 
       // Scan all non-CSS files for candidates
       transform(src, id, options) {
+        onFileChange(id)
+
         if (id.includes('/.vite/')) return
         let extension = getExtension(id)
         if (extension === '' || extension === 'css') return
 
-        scan(src, extension)
-        updateCssModules(options?.ssr ?? false)
+        let updated = scan(src, extension)
+        if (updated) {
+          queueRebuild(options?.ssr ?? false)
+        }
       },
     },
 
-    /*
-     * The plugins that generate CSS must run after 'enforce: pre' so @imports
-     * are expanded in transform.
-     */
-
     {
       // Step 2 (serve mode): Generate CSS
       name: '@tailwindcss/vite:generate:serve',
       apply: 'serve',
+      enforce: 'pre',
 
       async transform(src, id, options) {
         if (!isTailwindCssFile(id, src)) return
+        onFileChange(id)
 
         // In serve mode, we treat cssModules as a set, ignoring the value.
         cssModules[id] = { content: '', handled: true }
 
-        if (!options?.ssr) {
-          // Wait until all other files have been processed, so we can extract
-          // all candidates before generating CSS. This must not be called
-          // during SSR or it will block the server.
-          await server?.waitForRequestsIdle?.(id)
-        }
+        // if (!options?.ssr) {
+        //   // Wait until all other files have been processed, so we can extract
+        //   // all candidates before generating CSS. This must not be called
+        //   // during SSR or it will block the server.
+        //   await server?.waitForRequestsIdle?.(id)
+        // }
```

Comment:
Unclear why this is necessary. If any candidate change is detected, we already mark these files as updated anyways which will (if I’m not mistaken?) cause a retransform for this file anyways.